openstack/identity/v3/extensions/ec2tokens/requests.go

```diff
@@ -72,6 +72,9 @@ type AuthOptions struct {
 	BodyHash *string `json:"body_hash"`
 	// Timestamp is a timestamp to calculate a V4 signature. Optional.
 	Timestamp *time.Time `json:"-"`
+	// Token is a []byte string (encoded to base64 automatically) which was signed
+	// by an EC2 secret key. Used by S3 tokens for validation only.
+	Token []byte `json:"token,omitempty"`
```

Comment:
Just so I understand: this field is for someone to supply a pre-generated signed string?

Reply:
Exactly.